#! /bin/sh /usr/share/dpatch/dpatch-run
## http://bugzilla.kernel.org/attachment.cgi?id=16916
##
## DP: Fix double call to close_v4l2()

@DPATCH@
diff -urNad uvccapture-0.4~/uvccapture.c uvccapture-0.4/uvccapture.c
--- uvccapture-0.4~/uvccapture.c	2006-03-23 03:52:37.000000000 +0100
+++ uvccapture-0.4/uvccapture.c	2008-07-31 22:13:50.790128698 +0200
@@ -375,15 +375,16 @@
 	fclose (file);
 	videoIn->getPict = 0;
       }
-      if (post_capture_command[0])
+      if (post_capture_command[0]) {
 	if (verbose >= 1)
 	  fprintf (stderr, "Executing '%s %s'\n", post_capture_command[0],
 		   post_capture_command[1]);
-      if (spawn (post_capture_command, post_capture_command_wait, verbose)) {
-	fprintf (stderr, "Command exited with error\n");
-	close_v4l2 (videoIn);
-	free (videoIn);
-	exit (1);
+	if (spawn (post_capture_command, post_capture_command_wait, verbose)) {
+	  fprintf (stderr, "Command exited with error\n");
+	  close_v4l2 (videoIn);
+	  free (videoIn);
+	  exit (1);
+	}
       }
 
       ref_time = time (NULL);
